lambdas/credentialissuererror/src/main/java/uk/gov/di/ipv/core/credentialissuererror/CredentialIssuerErrorHandler.java

```diff
@@ -4,40 +4,53 @@
 import com.amazonaws.services.lambda.runtime.RequestHandler;
 import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
 import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+import com.nimbusds.oauth2.sdk.OAuth2Error;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import software.amazon.lambda.powertools.tracing.Tracing;
 import uk.gov.di.ipv.core.library.annotations.ExcludeFromGeneratedCoverageReport;
+import uk.gov.di.ipv.core.library.auditing.AuditEventTypes;
 import uk.gov.di.ipv.core.library.domain.JourneyResponse;
-import uk.gov.di.ipv.core.library.domain.UserStates;
 import uk.gov.di.ipv.core.library.dto.CredentialIssuerErrorDto;
+import uk.gov.di.ipv.core.library.exceptions.SqsException;
 import uk.gov.di.ipv.core.library.helpers.ApiGatewayResponseGenerator;
 import uk.gov.di.ipv.core.library.helpers.RequestHelper;
-import uk.gov.di.ipv.core.library.persistence.item.IpvSessionItem;
+import uk.gov.di.ipv.core.library.service.AuditService;
 import uk.gov.di.ipv.core.library.service.ConfigurationService;
-import uk.gov.di.ipv.core.library.service.IpvSessionService;
+
+import java.util.Arrays;
+import java.util.List;
 
 public class CredentialIssuerErrorHandler
         implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
 
     private static final Logger LOGGER =
             LoggerFactory.getLogger(CredentialIssuerErrorHandler.class.getName());
 
-    private static final String NEXT_JOURNEY_STEP_URI = "/journey/next";
+    private static final String ERROR_JOURNEY_STEP_URI = "/journey/error";
+
+    private static final List<String> ALLOWED_OAUTH_ERROR_CODES =
+            Arrays.asList(
+                    OAuth2Error.SERVER_ERROR_CODE,
+                    OAuth2Error.INVALID_REQUEST_CODE,
+                    OAuth2Error.INVALID_REQUEST_OBJECT_CODE,
+                    OAuth2Error.INVALID_GRANT_CODE,
+                    OAuth2Error.INVALID_CLIENT_CODE);
```

Comment:
What's the significance of this having this list? The OAuth spec lists a couple more responses, and the lib has loads more defined.

Reply:
Wasn't 100% sure about this to be honest as the ticket is fairly vague. This list is just the ones we have used so far from passport back. But maybe its better to update this list to be the full Oauth spec list?